org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/ast/ReferenceExpression.java

```diff
@@ -998,7 +998,20 @@ private ReferenceExpression cachedResolvedCopy(TypeBinding targetType) {
 			}
 			copy.setExpressionContext(this.expressionContext);
 			copy.setExpectedType(targetType);
+			// https://github.com/eclipse-jdt/eclipse.jdt.core/issues/1759
+			LocalVariableBinding[] locals = this.enclosingScope.locals;
+			if (locals != null) {
+				LocalVariableBinding[] copyLocals = new LocalVariableBinding[locals.length];
+				System.arraycopy(locals, 0, copyLocals, 0, locals.length);
+				this.enclosingScope.locals = copyLocals;
+				for (LocalVariableBinding local: copyLocals) {
+					if (local != null) {
+						local.modifiers = local.modifiers & (1 << ExtraCompilerModifiers.AccPatternVariable);
+					}
```

Comment:
@snjeza This seems to be a side effect of unknown nature. What exactly is attempted here ?? Why would you want to adjust the modifiers thus ??

Reply:
I agree. #1768 is better.